### PR TITLE
WW-4516: Allow JSON plug-in to skip bean cache when devMode is enabled.

### PR DESCRIPTION
**Pull request by Naros**
Adds support for disabling bean cache on JSON plugin when devMode is enabled.

### CODE REVIEW

File: plugins/json/src/main/java/org/apache/struts2/json/JSONResult.java [hunk 1]
```diff
@@ -92,12 +96,18 @@
     private String contentType;
     private String wrapPrefix;
     private String wrapSuffix;
-
+    private boolean debugMode = false;
```
Comment (lukaszlenart) [inline plugins/json/src/main/java/org/apache/struts2/json/JSONResult.java]: Please rename this to `devMode`

File: plugins/json/src/main/java/org/apache/struts2/json/JSONUtil.java [hunk 2]
```diff
@@ -53,7 +68,22 @@
      * @throws JSONException
      */
     public static String serialize(Object object) throws JSONException {
```
Comment (lukaszlenart) [inline plugins/json/src/main/java/org/apache/struts2/json/JSONUtil.java]: Is this method used somewhere else? If not, it's the perfect time to drop it

Comment (Naros) [inline plugins/json/src/main/java/org/apache/struts2/json/JSONUtil.java]: Not that I can see from just scanning my github fork.  I'll double check when I get home tonight and if not; I'll remove it.

Comment (Naros) [inline plugins/json/src/main/java/org/apache/struts2/json/JSONUtil.java]: It is currently used in the test cases
JSONResultTest - testJSONUtilNPEOnNullMethod()
JSONUtilTest = testSerializeDeserialize()
Thoughts?

Comment (lukaszlenart) [inline plugins/json/src/main/java/org/apache/struts2/json/JSONUtil.java]: Throw away and use the new version

File: plugins/json/src/main/java/org/apache/struts2/json/JSONResult.java (hunk 1 above)
Comment (Naros) [inline plugins/json/src/main/java/org/apache/struts2/json/JSONResult.java]: Changed in latest commits per your request.

File: plugins/json/src/main/java/org/apache/struts2/json/JSONResult.java [hunk 3]
```diff
@@ -92,12 +96,18 @@
...
+    private boolean devMode = false;
+    
     @Inject(StrutsConstants.STRUTS_I18N_ENCODING)
     public void setDefaultEncoding(String val) {
         this.defaultEncoding = val;
     }
-
+    
+    @Inject(StrutsConstants.STRUTS_DEVMODE) 
+    public void setDebugMode(String val) {
```
Comment (lukaszlenart) [inline plugins/json/src/main/java/org/apache/struts2/json/JSONResult.java]: Can you rename setter as well? Thanks!

**Pull request merged**